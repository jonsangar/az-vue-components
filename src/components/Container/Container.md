# Contenedor

Un contenedor envuelve nuestro diseño y nos permite definir si queremos que ocupe todo el ancho disponible o que se ajuste a un máximo.

Este componente no aporta ninguna funcionalidad, solo simplifica el manejo del framework evitando tener que utilizar varias clases. Gracias a las propiedades, podemos adaptar el diseño de forma dinámica si fuese necesario.

**Por defecto el framework base incorpora todas las clases que utiliza este componente.**

### Instalación

```javascript
import { Container } from 'didor';

Vue.use(Container);
```

---

### Uso

#### Predefinido

Por defecto el componente `container` limita el ancho máximo de su contenido, según los valores configurados para cada tamaño de pantalla.

::: demo
<az-container>
  <p>
    Lorem ipsum dolor sit amet, consectetur adipiscing elit. Duis euismod elementum enim, ac aliquet sem consectetur non. Integer suscipit lacus eu turpis sagittis, sit amet venenatis leo dignissim. Suspendisse facilisis urna dolor, sit amet lacinia dui mollis id. Curabitur tincidunt varius purus, a scelerisque nibh accumsan in. Ut iaculis ligula mi, sed hendrerit ante vestibulum in. Mauris id pharetra sapien, vel semper tellus. Mauris gravida porttitor purus at viverra. Praesent at elementum elit. Integer in convallis tortor. Phasellus vel auctor enim, sit amet mollis odio. Nulla at fringilla magna, in mattis neque. Phasellus vitae leo at nibh fermentum luctus. Aliquam erat volutpat.
  </p>
</az-container>
:::

```html
<az-container>
  <p>Lorem ipsum dolor sit amet, consectetur adipiscing elit...</p>
</az-container>
```

#### Fluido

Si queremos que nuestro diseño se adapte al 100% del ancho disponible, solo tenemos que pasarle el atributo `fluid`.

::: demo
<az-container fluid>
  <p>
    Lorem ipsum dolor sit amet, consectetur adipiscing elit. Duis euismod elementum enim, ac aliquet sem consectetur non. Integer suscipit lacus eu turpis sagittis, sit amet venenatis leo dignissim. Suspendisse facilisis urna dolor, sit amet lacinia dui mollis id. Curabitur tincidunt varius purus, a scelerisque nibh accumsan in. Ut iaculis ligula mi, sed hendrerit ante vestibulum in. Mauris id pharetra sapien, vel semper tellus. Mauris gravida porttitor purus at viverra. Praesent at elementum elit. Integer in convallis tortor. Phasellus vel auctor enim, sit amet mollis odio. Nulla at fringilla magna, in mattis neque. Phasellus vitae leo at nibh fermentum luctus. Aliquam erat volutpat.
  </p>
</az-container>
:::

``` html
<az-container fluid>
  <p>Lorem ipsum dolor sit amet, consectetur adipiscing elit...</p>
</az-container>
```

### API

| Attribute   | Description                                | Type      | Default   |
| ----------- | ------------------------------------------ | --------- | --------- |
| fluid        | Determina como debe de adaptarse su contenido | `Boolean`  | `false` |
