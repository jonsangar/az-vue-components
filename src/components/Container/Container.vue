<template lang="pug">
div(:class="classObject")
  slot
</template>

<script>
export default {
  props: {
    fluid: {
      type: Boolean,
      default: false,
    },
  },
  computed: {
    classObject() {
      return [this.fluid ? 'container-fluid' : 'container'];
    },
  },
};
</script>
