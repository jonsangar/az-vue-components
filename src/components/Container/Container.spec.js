import { shallowMount, createLocalVue } from '@vue/test-utils';
// Importamos librerÃ­as de estilos e iconos
import SvgIcon from 'vue-svgicon'
import Container from './Container.vue';

const localVue = createLocalVue()
localVue.use(SvgIcon, {
  tagName: 'az-icon',
  classPrefix: 'az',
})

describe('Container.vue', () => {
  const build = () => {
    const wrapper = shallowMount(Container, {
      localVue,
    })

    return {
      wrapper,
    }
  }

  it('Comprueba renderizado del componente Container con snapshot', () => {
    const { wrapper } = build()
    expect(wrapper.html()).toMatchSnapshot()

  })
})
