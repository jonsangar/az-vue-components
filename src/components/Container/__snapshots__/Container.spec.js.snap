// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Container.vue Comprueba renderizado del componente Container con snapshot 1`] = `<div class="container"></div>`;
